sentry-types/src/protocol/profile.rs

```diff
@@ -1,77 +1,143 @@
+use std::collections::HashMap;
+
 use super::v7::{DebugMeta, TraceId};
+
+use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize, Serializer};
 use uuid::Uuid;
 
 fn serialize_id<S: Serializer>(uuid: &Uuid, serializer: S) -> Result<S::Ok, S::Error> {
     serializer.serialize_some(&uuid.as_simple())
 }
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Metadata about the transaction associated with the profile
+pub struct TransactionMetadata {
+    #[serde(serialize_with = "serialize_id")]
+    /// Transaction ID
+    pub id: Uuid,
+    /// Transaction Name
+    pub name: String,
+    /// Trace ID
+    pub trace_id: TraceId,
+    /// Transaction start timestamp in nanoseconds
+    pub relative_start_ns: u64,
+    /// Transaction end timestamp in nanoseconds
+    pub relative_end_ns: u64,
+    /// ID of the thread in which the transaction started
+    #[serde(default)]
+    pub active_thread_id: u64,
+}
 
-#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
-/// Represents a Symbol
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Single frame of a Sample
 pub struct RustFrame {
-    /// Raw instruction address
+    /// Instruction address
     pub instruction_addr: String,
 }
 
-#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
-/// Represents a Sample
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Single sample of a profile
 pub struct Sample {
-    /// List of symbols
-    pub frames: Vec<RustFrame>,
-    /// The thread name
-    pub thread_name: String,
-    /// The thread id
+    /// ID of the relative stack
+    pub stack_id: u32,
+    /// Thread ID
     pub thread_id: u64,
-    /// Nanoseconds elapsed between when the profiler started and when this sample was collected
-    pub nanos_relative_to_start: u64,
+    /// Timestamp at which this sample was collected
+    pub relative_timestamp_ns: u64,
```

Comment:
relative to what? the start of the profiling session, or relative to the transaction start?

Reply:
as above: good point, let me add that in the comments!